tests/system/src/test/java/test/com/sun/javafx/font/freetype/PangoTest.java

```diff
@@ -0,0 +1,147 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.com.sun.javafx.font.freetype;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Scene;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.VBox;
+import javafx.scene.text.Text;
+import javafx.stage.Stage;
+
+import org.junit.Test;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+
+import junit.framework.AssertionFailedError;
+import static test.util.Util.TIMEOUT;
+
+import static org.junit.Assert.*;
+
+/**
+ * Test program for UTF16 to UTF8 conversion and Pango
+ */
+public class PangoTest {
+
+    static CountDownLatch launchLatch = new CountDownLatch(1);
+
+    static MyApp myApp;
+    static Pane pane;
+
+    public static class MyApp extends Application {
+
+        Stage stage = null;
+
+        public MyApp() {
+            super();
+        }
+
+        @Override
+        public void init() {
+            myApp = this;
+        }
+
+        @Override
+        public void start(Stage primaryStage) throws Exception {
+            this.stage = primaryStage;
+            pane = new VBox(10);
+            Scene scene = new Scene(pane, 400, 200);
+            stage.setScene(scene);
+            stage.show();
+            launchLatch.countDown();
+        }
+    }
+
+    @BeforeClass
+    public static void setupOnce() {
+        // Start the Application
+        new Thread(() -> Application.launch(MyApp.class, (String[]) null)).start();
+
+        try {
+            if (!launchLatch.await(TIMEOUT, TimeUnit.MILLISECONDS)) {
+                throw new AssertionFailedError("Timeout waiting for Application to launch");
+            }
+        } catch (InterruptedException ex) {
+            AssertionFailedError err = new AssertionFailedError("Unexpected exception");
+            err.initCause(ex);
+            throw err;
+        }
+
+        assertEquals(0, launchLatch.getCount());
+    }
+
+
+
+    @AfterClass
+    public static void teardownOnce() {
+        Platform.exit();
+    }
+
+    private void addTextToPane(Text text) {
+        final CountDownLatch rDone = new CountDownLatch(1);
+        Platform.runLater(() -> {
+            text.layoutYProperty().addListener(inv -> {
+                rDone.countDown();
+            });
+            pane.getChildren().add(text);
+        });
+
+        try {
+            if (!rDone.await(TIMEOUT, TimeUnit.MILLISECONDS)) {
+                throw new AssertionFailedError("Timeout waiting for runLater");
```

Comment:
This could also be replaced with `assertTrue("Timeout ...", rDone.await(TIMEOUT, TimeUnit.MILLISECONDS)` if you add `throws Exception` to this method and the calling test methods.